### PR TITLE
detect dotnet version

### PR DESCRIPTION
**Pull request by vashian**
Personally, I don't like to use the `child_process` module.
If it's not clear, I'm going to change it and use `dotnet --info` to detect the version.

### CODE REVIEW

**Comment by vashian** [comment]
I found in the `.csproj` file we can specify versions in different ways like: `appnetcore3.0` or `net6.0`
Need to update **REGEX**

**Comment by mhemrg** [comment]
I think using `dotnet --info` is more reliable than this approach.

**Comment by vashian** [comment]
Here's the result of the command: `dotnet --version`
image

maybe users don't install it, or maybe just want to deploy their package with a different version, any idea?

**Comment by vashian** [comment]
What's the state of this **PR**?

**Comment by mhemrg** [comment]
I was wrong about this. Your first approach was correct. Reading the version from the `.csproj` file is the right way to do this.

Please remove the `getAllFiles` function. I just don't like it. It may slow users' computer. 

Please take a look at the `detect-platform.ts` file. Use the following code to find the `.csproj` file and then use regex to find out the version:

```javascript
  const [programCSFilePath] = globby.sync('**/{Startup.cs,Program.cs}', {
    cwd: projectPath,
    gitignore: true,
    deep: 5,
  });

  const [CSProjFile] = programCSFilePath && globby.sync('*.csproj', {
    gitignore: true,
    cwd: path.join(projectPath, path.dirname(programCSFilePath)),
  });
```

File: src/utils/merge-platform-config.ts [hunk 1]
```diff
@@ -1,21 +1,79 @@
 import path from 'path'
+
 import fs from 'fs-extra'
 import semver from 'semver'
+import findVersions from 'find-versions'
```
Comment (mhemrg) [inline src/utils/merge-platform-config.ts:5]: We don't need this line anymore.

File: src/utils/list-file-recursive.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import path from "path";
```
Comment (mhemrg) [inline src/utils/list-file-recursive.ts:1]: You should delete the whole file.

**Pull request merged**